src/transform3d.rs

```diff
@@ -652,6 +653,31 @@ impl<T: Copy, Src, Dst> TypedTransform3D<T, Src, Dst> {
     }
 }
 
+impl<T0: NumCast + Copy, Src, Dst> TypedTransform3D<T0, Src, Dst> {
+    /// Cast from one numeric representation to another, preserving the units.
+    pub fn cast<T1: NumCast + Copy>(&self) -> Option<TypedTransform3D<T1, Src, Dst>> {
+        match (NumCast::from(self.m11), NumCast::from(self.m12),
```

Comment:
I wonder if it would be easier somehow to extract the vectors from a matrix, cast them (it's already there), and compose a new matrix from vectors